File: gcd/Gcd.scala
import  scala.annotation.tailrec

object Gcd {
  def main(args: Array[String]): Unit = {
    if (args.length != 2) {
      println("Provide 2 numbers as input to the program")
      println("Usage: scala Gcd.scala <num1> <num2>")
      System.exit(1)
    }
    else {
      println("GCD is "+ gcd(args(0).toInt,args(1).toInt))
    }
  }

  @tailrec
  def gcd(num1: Int, num2: Int) : Int = {
    if(num2 == 0) num1 else gcd(num2 , num1 % num2)
  }
}

File: gcd/GCD.go
package main
import (
        "fmt"
)

func gcd(num1 int, num2 int) int {
    for num2 != 0 {
            temp := num2
            num2 = num1 % num2
            num1 = temp
    }
    return num1
}

func main() {
    var num1, num2 int
    fmt.Println("Enter two numbers: ")
    fmt.Scanln(&num1)
    fmt.Scanln(&num2)

    fmt.Printf("gcd of %d and %d is %d",num1,num2,gcd(num1, num2))
}
